plugins/tiddlywiki/menubar/menu.tid

```diff
@@ -7,34 +7,37 @@ tags: $:/tags/PageTemplate
 <$list filter="[<currentTiddler>addprefix[$:/config/plugins/menubar/MenuItems/Visibility/]get[text]] ~show +[match[show]]" variable="ignore">
 <$list filter="[[$size$]match[wide]] ~[<currentTiddler>get[show-when]match[$size$]] ~[{$:/state/popup/menubar/hamburger}match[yes]]" variable="ignore">
 <li style={{!!custom-menu-styles-$size$}} class={{{ [<currentTiddler>get[show-when]addprefix[tc-menubar-]] tc-menubar-item +[join[ ]] }}}>
-<$list filter="[<currentTiddler>!is-dropdown[yes]]" variable="listItem" emptyMessage="""
+<%if [<currentTiddler>!is-dropdown[yes]] %>
+<%if [<currentTiddler>has[custom-menu-content]] %>
+	<$transclude field="custom-menu-content" mode="inline"/>
+<%else%>
+	<!-- Link -->
+	<$link to={{!!target}}>
+	<$set name="tv-wikilinks" value="no">
+	<$transclude field="caption" mode="inline"/>
+	</$set>
+	</$link>
+<%endif%>
+<%else%>
 	<!-- Dropdown -->
 	<$set name="dropdown-state" value=<<qualify "$:/state/popup/topmenu/dropdown/">>>
 	<$set name="dropdown-state" value={{{ [<dropdown-state>addsuffix<currentTiddler>] }}}>
 	<$button popup=<<dropdown-state>> selectedClass="tc-selected">
+	<%if [<currentTiddler>has[custom-menu-content]] %>
+	<$transclude field="custom-menu-content" mode="inline"/>
+	<%else%>
 	<$set name="tv-wikilinks" value="no">
 	<$transclude field="caption" mode="inline"/>
```

Comment:
If there is no caption field, it should show the title.

Reply:
Is that what the old code does?  When I create a tiddler with no `caption`, I see a menubar item with no text in it.